src/Api/Endpoints/Authenticators.cs

```diff
@@ -0,0 +1,73 @@
+using Microsoft.AspNetCore.Mvc;
+using Passwordless.Api.Authorization;
+using Passwordless.Common.Models.Authenticators;
+using Passwordless.Service;
+using Passwordless.Service.Features;
+using Passwordless.Service.Helpers;
+using static Microsoft.AspNetCore.Http.Results;
+
+namespace Passwordless.Api.Endpoints;
+
+public static class AuthenticatorsEndpoints
+{
+    public static void MapAuthenticatorsEndpoints(this WebApplication app)
+    {
+        var group = app.MapGroup("/authenticators")
+            .RequireCors("default");
+
+        group.MapGet("/list", ListConfiguredAuthenticatorsAsync)
+            .RequireSecretKey();
+
+        group.MapPost("/whitelist", WhitelistAuthenticatorsAsync)
+            .WithParameterValidation()
+            .RequireSecretKey();
+
+        group.MapPost("/delist", DelistAuthenticatorsAsync)
+            .WithParameterValidation()
+            .RequireSecretKey();
+    }
+
+    public static async Task<IResult> ListConfiguredAuthenticatorsAsync(
```

Comment:
Nit: I'd personally inline the methods into delegates, both for consistency and because they're quite small anyway